### PR TITLE
Add datasync PyPi dependency

### PR DESCRIPTION
**Pull request by eightysteele**
This PR add a dependency on our new `datasync` PyPi dependency that is now published to Google Artifact Registry.

## Before merging 
- [x] Switch the branch in `.github/workflows/pixi.yml` back to `main`

### CODE REVIEW

File: .github/workflows/pixi.yml [hunk 1]
```diff
@@ -4,14 +4,22 @@ on:
...
-      - main
+      - eightysteele/gen-469-integrate-datasync-package-with-b3d
 
 jobs:
   pixi-update:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
 
+      # - name: Checkout probcomp/datasync
```
Comment (sritchie) [inline .github/workflows/pixi.yml:15]: let's remove the comment block

File: .github/workflows/pixi.yml [hunk 2]
```diff
@@ -25,7 +33,8 @@ jobs:
         run: |
           # install pixi
           curl -fsSL https://pixi.sh/install.sh | bash
-          export PATH=/home/runner/.pixi/bin:$PATH
+          export PATH=/home/runner/.pixi/bin:/home/runner/.local/bin:$PATH
```
[the remark this answers is not in the document]
Comment (eightysteele) [reply .github/workflows/pixi.yml:36]: It just updates `PATH` to include `pipx` (needed for `keyring` during GCP authentication).

File: pyproject.toml [hunk 3]
```diff
@@ -109,7 +109,8 @@ cwd = "scripts"
 
 [tool.pixi.feature.core.pypi-dependencies]
 carvekit = "==4.1.2"
-genjax = "*"
+datasync = "*"
```
[the remark this answers is not in the document]
Comment (eightysteele) [reply pyproject.toml:112]: For sure! Are there strong opinions for pinning vs. wildcards? Here's my thought process:

- Good default that's flexible and can be overridden if needed
- Supports automatic updates (when triggered) compatible across all deps, environments ,and platforms
- Exact package versions get pinned automatically in the lock file for reproducibility

The main downside is a risk of breaking changes from a weird update, which I haven’t seen yet. But in theory, that should be caught in CI and then we can just rollback the lock file.

Either way is fine though! Let me know and I can send PRs to get all versions locked down vs. wildcards.

**Comment by eightysteele** [comment]
@sritchie thanks for the look, pushed changes.

**Pull request merged**